plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java

```diff
@@ -71,6 +71,10 @@ public void configure(Binder binder)
         binder.bind(JdbcClient.class).to(CachingJdbcClient.class).in(Scopes.SINGLETON);
 
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class));
+        newOptionalBinder(binder, JdbcIdentityCacheMapping.class)
+                .setDefault()
+                .to(SingletonJdbcIdentityCacheMapping.class)
```

Comment:
It would be safer not to bind it by default.
It's very related to `CredentialProvider` interface (which has no default binding), so we could bind proper `JdbcIdentityCacheMapping` when binding that guy.